lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -66,6 +68,23 @@ def object? object
       object.type.include? RDFVocabularies::PCDMTerms.Object
     end
 
+    # TODO: This is ugly. Are there clever ways to assemble a list of ancestors for a 
+    # given collection?
+    def collection_ancestor? collections
+      next_batch = []
+      current_collections = collections
+      while !current_collections.empty? do
+        current_collections.each do |check|
+          check.collections.each do |c|
+            return true if ActiveFedora::Base.find(c.id).collections.map { |x| x.id == self.id }
```

Comment:
@flyingzumwalt yes, it will be slow. At this point, I don't see a solution, unless we leverage solr somehow, or have an ancestors method that we could call on an existing PCDM::Collection or PCDM::Object that would return a list of its ancestor ids or Ruby objects.